src/backbone.virtual-collection.js

```diff
@@ -117,6 +117,7 @@ var VirtualCollection = Backbone.VirtualCollection = Backbone.Collection.extend(
         this.trigger('change', model, this, options);
       } else {
         this._indexAdd(model);
+        this.listenTo(model, 'all', this._onAllEvent);
```

Comment:
can you make this else be a call to `this._onAdd(model, this.collection, options)` so that the changes cache stays in sync?